### DISCUSSION TITLE
Conditional import of Capacitor plugins?

### DISCUSSION THREAD

**Discussion by Dylan190774** [quasarframework/quasar#15375] [CLI - Capacitor / Cordova modes]
I'm creating an app that runs in the browser, but can also be packed into a Capacitor app.
I use the `@capacitor/camera` plugin, which needs the following import in a component that uses the camera:

`import { Camera, CameraResultType } from '@capacitor/camera';`

Since the camera plugin is only installed in `src-capacitor`  (not in `package.json` in the root of the project), this gives an error when running the app in a browser.

Is there any way to make the import conditional?
I use Vite.

**Comment by MilosPaunovic** [comment]
You might want to have a look at platform for conditional imports:
https://quasar.dev/options/platform-detection

**Comment by metalsadman** [comment]
see this https://quasar.dev/quasar-cli-vite/handling-process-env#example 2nd example. ie `const {Camera, CameraResultType} = await import('app/src-capacitor/node_modules/@capacitor/camera')`, or register an alias to your src-cpacitor/node_modules folder.
